client/starwhale/core/runtime/model.py

```diff
@@ -177,37 +183,289 @@ def __str__(self) -> str:
     __repr__ = __str__
 
 
+class BaseDependency(Protocol):
+    def __init__(self, deps: t.Any) -> None:
+        ...
+
+    @property
+    def kind(self) -> DependencyType:
+        ...
+
+    def conda_install(self, workdir: Path, env_dir: Path, configs: t.Dict) -> None:
+        ...
+
+    def venv_install(self, workdir: Path, env_dir: Path, configs: t.Dict) -> None:
+        ...
+
+    def asdict(self, ignore_keys: t.Optional[t.List[str]] = None) -> t.Dict:
+        ...
+
+
+class NativeFileDependency(ASDictMixin, BaseDependency):
+    def __init__(self, deps: t.List[t.Dict[str, str]]) -> None:
+        if isinstance(deps, list):
+            for d in deps:
+                if not isinstance(d, dict):
+                    raise FormatError(f"native file must dict type: {d}")
+                if not d.get("src") or not d.get("dest"):
+                    raise FormatError(
+                        "dependencies.file MUST include src and dest fields."
+                    )
+        else:
+            raise NoSupportError(
+                f"native file dependency only support list[dict]: {deps}"
+            )
+        self.deps = deps
+
+    @property
+    def kind(self) -> DependencyType:
+        return DependencyType.NATIVE_FILE
+
+    def _do_install(self, workdir: Path, mode: str) -> None:
+        # TODO: support native file pre-hook, post-hook
+        for d in self.deps:
+            if d["dest"].startswith("/") and in_container():
+                logger.warning(f"NOTICE! dest dir: {d['dest']}")
```

Comment:
Get the absolute path and then compare the commonprefix with work dir?
`dest` might be "../../../"

Reply:
ok

Reply:
fixed